extensions/oidc/runtime/src/main/java/io/quarkus/oidc/runtime/CodeAuthenticationMechanism.java

```diff
@@ -1065,23 +1065,35 @@ public Uni<? extends Void> apply(Void t) {
                             LOG.error(error);
                             throw new AuthenticationCompletionException(error);
                         }
-                        long maxAge = idTokenJson.getLong("exp") - idTokenJson.getLong("iat");
-                        LOG.debugf("ID token is valid for %d seconds", maxAge);
+                        long idTokenAge = idTokenJson.getLong("exp") - idTokenJson.getLong("iat");
+                        LOG.debugf("Session age is initialized with ID token age of %d seconds", idTokenAge);
+                        long sessionAge = idTokenAge;
                         if (configContext.oidcConfig().token().lifespanGrace().isPresent()) {
-                            maxAge += configContext.oidcConfig().token().lifespanGrace().getAsInt();
+                            int lifespanGrace = configContext.oidcConfig().token().lifespanGrace().getAsInt();
+                            LOG.debugf("Adding token lifespan grace of %d seconds to the session age", lifespanGrace);
+                            sessionAge += lifespanGrace;
                         }
-                        if (configContext.oidcConfig().token().refreshExpired() && tokens.getRefreshToken() != null) {
-                            maxAge += configContext.oidcConfig().authentication().sessionAgeExtension().getSeconds();
+                        if (configContext.oidcConfig().token().refreshExpired()) {
+                            if (tokens.getRefreshToken() != null) {
+                                long sessionAgeExtension = configContext.oidcConfig().authentication().sessionAgeExtension()
+                                        .orElse(Duration.ofMinutes(5)).getSeconds();
```

Comment:
I honestly don't see how making it `Optional` changes anything, you're still expecting a value here and defaulting to 5min.

Reply:
Hi @gastaldi, if it set by default, we can't detect if a user cares at all about extending the session age, 5 mins makes no real difference, while enabling a token refresh is a policy decision (without it a user has to re-authenticate when the session expires, with it it can last for a very long time), so in those cases users may get surprised to see a warning...

Does it clarify it ?

Reply:
I.e, we can keep it as it is, with a default value, but then we won't be able to warn that a user configured session extension won't work unless the tokens can be refreshed

Reply:
> I.e, we can keep it as it is, with a default value, but then we won't be able to warn that a user configured session extension won't work unless the tokens can be refreshed

Maybe I am missing something,  but can you point me to where you check if the `configContext.oidcConfig().authentication().sessionAgeExtension()` is present?